applications/DamApplication/custom_strategies/schemes/bossak_displacement_smoothing_scheme.hpp

```diff
@@ -57,6 +57,21 @@ class BossakDisplacementSmoothingScheme : public ResidualBasedBossakDisplacement
         r_model_part.GetProcessInfo()[RAYLEIGH_ALPHA] = mRayleighAlpha;
         r_model_part.GetProcessInfo()[RAYLEIGH_BETA] = mRayleighBeta;
 
+        const int NNodes = static_cast<int>(r_model_part.Nodes().size());
+        ModelPart::NodesContainerType::iterator node_begin = r_model_part.NodesBegin();
+
+        // Initialize INITIAL_STRESS_TENSOR
+        #pragma omp parallel for
+        for(int i = 0; i < NNodes; i++)
+        {
+            ModelPart::NodesContainerType::iterator itNode = node_begin + i;
+
+            Matrix& rInitialStress = itNode->FastGetSolutionStepValue(INITIAL_STRESS_TENSOR);
+            if(rInitialStress.size1() != 3)
+                rInitialStress.resize(3,3,false);
+            noalias(rInitialStress) = ZeroMatrix(3,3);
```

Comment:
You're always allocating the memory by calling the `ZeroMatrix` so you can skip the size check. Alternatively, you could call the clear to initialize it without reallocating the memory (provided of course that the sizes are OK).

Reply:
PS: We are calling ZeroMatrix but we are using noalias and, if I am not mistaken, the variable is not redefined and neither its size

Reply:
For the first allocation you're right. For the succeeding ones I think I am. The point is that as far as I remember, the `ZeroMatrix` creates a temporary that is assigned to `rInitialStress` (I might be mistaken of course). My point is that I'd have done something like
```c++
if (rInitialStress.size1() != 3 || rInitialStress.size2() != 3) {
    rInitialStress.resize(3,3,false);
}
rInitialStress.clear();
``
Nevertheless, it's a tiny detail.

Reply:
Ok, I see. Thank you Ruben, we will try this solution.

Reply:
Thank you, @xaviertort and @ipouplana for the work and @rubenzorrilla for the review.

I have just added some tests for the joints in the Dam Application. We will use the elastic cohesive and the isotropic damage cohesive joints, they are robust enough for our problems.